csharp/src/Apache.Arrow.Adbc/AdbcStatement.cs

```diff
@@ -0,0 +1,154 @@
+        {
+            return await Task.Run(() => ExecuteQuery());
+        }
+
+        /// <summary>
+        /// Executes an update command and returns the number of records effected.
+        /// </summary>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public abstract UpdateResult ExecuteUpdate();
+
+        // <summary>
+        /// Executes an update command and returns the number of records effected.
+        /// </summary>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public virtual async Task<UpdateResult> ExecuteUpdateAsync()
+        {
+            return await Task.Run(() => ExecuteUpdate());
+        }
+
+        /// <summary>
+        /// Execute a result set-generating query and get a list of partitions of the result set.
+        /// </summary>
+        /// <returns><see cref="PartitionedResult"/></returns>
+        public virtual PartitionedResult ExecutePartitioned()
+        {
+            throw AdbcException.NotImplemented("Statement does not support ExecutePartitioned");
+        }
+
+        /// <summary>
+        /// Get the schema for bound parameters.
+        /// </summary>
+        /// <returns><see cref="Schema"/></returns>
+        public virtual Schema GetParameterSchema()
+        {
+            throw AdbcException.NotImplemented("Statement does not support GetParameterSchema");
+        }
+
+        /// <summary>
+        ///  Turn this statement into a prepared statement to be
+        ///  executed multiple times.
+        /// </summary>
+        public virtual void Prepare()
+        {
+            throw AdbcException.NotImplemented("Statement does not support Prepare");
+        }
+
+        public virtual void Dispose()
+        {
+        }
+
+        /// <summary>
+        /// Gets a value from the Arrow array at the specified index, using the Field metadata for information.
+        /// </summary>
+        /// <param name="arrowArray">The Arrow array.</param>
+        /// <param name="field">The <see cref="Field"/> from the <see cref="Schema"/> that can be used for metadata inspection.</param>
+        /// <param name="index">The index in the array to get the value from.</param>
+        /// <returns></returns>
+        public abstract object GetValue(IArrowArray arrowArray, Field field, int index);
+
+        /// <summary>
+        /// For decimals, Arrow throws an OverflowException if a value is < decimal.min or > decimal.max
+        /// So parse the numeric value and return it as a string, if possible
+        /// </summary>
+        /// <param name="oex"></param>
+        /// <returns>A string value of the decimal that threw the exception or rethrows the OverflowException.</returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public virtual string ParseDecimalValueFromOverflowException(OverflowException oex)

```

Comment:
This still doesn't feel like it belongs in the public API, at least not as part of a Statement object